c_emulator/riscv_platform.c

```diff
@@ -47,6 +47,21 @@ bool sys_enable_vext(unit u)
   return rv_enable_vext;
 }
 
+uint64_t sys_pmp_count(unit u)
+{
+  return rv_pmp_count;
+}
+
+uint64_t sys_pmp_writable(unit u)
+{
+  return rv_pmp_writable;
```

Comment:
From a spec point of view, "writable" is not a boolean. Each field in pmpcfg could be writable or not, and which values they can hold if they are writable is configuration option (e.g. each pmpconfig may support TOR or NAPOT or both, and if pmpaddr or pmpcfg isn't writable, the value they hold is not necessarily zero (and that was added to the spec for specific security use cases). This is a good first step, but certainly isn't complete

Reply:
Yep, like all the WARL stuff in the model, this only allows a small amount of configuration compared to the near-infinite possibilities that the spec allows. I really only added it because it's needed for our chip, but maybe I will just remove it from this PR since it will (hopefully) eventually be replaced by the `riscv-config` WARL expressions anyway, and it will make reviewing this a bit easier.